distribution/openrct2.d.ts

```diff
@@ -854,6 +854,123 @@ declare global {
         remove(): void;
     }
 
+    /**
+     * Represents a single car on a ride. A train is made up of multiple cars, but
+     * something like boat hire will be one car per boat.
+     */
+    interface Car extends Entity {
+        /**
+         * The ride this car belongs to.
+         */
+        ride: number;
+
+        /**
+         * The ride object for this car, e.g. the ladybird trains object.
+         */
+        rideObject: number;
+
+        /**
+         * The vehicle type for the ride object used. This is a local index
+         * into the ride object list of vehicle types.
+         */
+        vehicleObject: number;
+
+        spriteType: number;
+
+        /**
+         * How many seats the car has, i.e. the capacity.
+         */
+        numSeats: number;
+
+        /**
+         * The next car on the same train. If this is the last or only car on the train,
+         * this will return null.
+         */
+        nextCarOnTrain: number | null;
+
+        /**
+         * The previous car on the ride. This may be the on the same train or the previous
+         * train. This will point to the last car if this is the first car on the ride.
+         */
+        previousCarOnRide: number;
+
+        /**
+         * The next car on the ride. This may be the on the same train or the next
+         * train. This will point to the first car if this is the last car on the ride.
+         */
+        nextCarOnRide: number;
+
+        /**
+         * The current station the train is in or departing.
+         */
+        currentStation: number;
+
+        /**
+         * How heavy the car is. This is the sum of the mass of the empty car and the
+         * mass of each guest that is riding it.
+         */
+        mass: number;
+
+        /**
+         * How much the car's velocity changes per tick.
+         */
+        acceleration: number;
+
+        /**
+         * How fast the car is moving.
+         */
+        velocity: number;
+
+        /**
+         * The current tilt of the car in the X/Y axis.
+         */
+        bankRotation: number;
+
+        /**
+         * The colour of the car.
+         */
+        colours: VehicleColour;
+
+        speed: number;
+        poweredAcceleration: number;
```

Comment:
I feel like this one needs a comment (there are others more obvious that don't)

Reply:
Any that don't have a comment is because I can't think of a good comment. Suggestions welcome.

Reply:
Does this means the acceleration at launch?
`/* Acceleration value to be used when the train exits the station */`

Reply:
@duncanspumpkin do you agree?

Reply:
Pretty sure powered acceleration is the field for vehicles that are self powered. Such as steam trains, boats, ...

Reply:
What is speed then?

Reply:
speed is the max speed that can be obtained from powered acceleration. We should rename it so that it matches the vehicle object name `powered_max_speed`